platform/commonUI/browse/test/ObjectHeaderControllerSpec.js

```diff
@@ -0,0 +1,118 @@
+                mockMutationCapability,
+                mockTypeCapability,
+                mockEvent,
+                mockCurrentTarget,
+                controller;
+
+            beforeEach(function () {
+                mockMutationCapability = jasmine.createSpyObj("mutation", ["mutate"]);
+                mockTypeCapability = {
+                    typeDef: {
+                        name: ""
+                    }
+                };
+                mockCapabilities = {
+                    mutation: mockMutationCapability,
+                    type: mockTypeCapability
+                };
+
+                mockDomainObject = jasmine.createSpyObj("domainObject", ["getCapability", "model"]);
+                mockDomainObject.model = {name: "Test name"};
+                mockDomainObject.getCapability.andCallFake(function (key) {
+                    return mockCapabilities[key];
+                });
+
+                mockScope = {
+                    domainObject: mockDomainObject
+                };
+
+                mockCurrentTarget = jasmine.createSpyObj("currentTarget", ["blur", "innerHTML"]);
+                mockCurrentTarget.blur.andReturn(mockCurrentTarget);
+
+                mockEvent = {
+                    which: {},
+                    type: {},
+                    currentTarget: mockCurrentTarget
+                };
+
+                controller = new ObjectHeaderController(mockScope);
+            });
+
+            it("updates the model with new name on blur", function () {
+                mockEvent.type = "blur";
+                mockCurrentTarget.innerHTML = "New name";
+                controller.updateName(mockEvent);
+
+                expect(mockMutationCapability.mutate).toHaveBeenCalled();
+            });
+
+            it("updates the model with a default for blank names", function () {
+                mockEvent.type = "blur";
+                mockCurrentTarget.innerHTML = "";
+                controller.updateName(mockEvent);
+
+                expect(mockCurrentTarget.innerHTML.length).not.toEqual(0);
+                expect(mockMutationCapability.mutate).toHaveBeenCalled();
+            });
+
+            it("does not update the model if the same name", function () {
+                mockEvent.type = "blur";
+                mockCurrentTarget.innerHTML = mockDomainObject.model.name;
+                controller.updateName(mockEvent);
+
+                expect(mockMutationCapability.mutate).not.toHaveBeenCalled();
+            });
+
+            it("updates the model on enter keypress event only", function () {
+                mockCurrentTarget.innerHTML = "New name";
+                controller.updateName(mockEvent);
+
+                expect(mockMutationCapability.mutate).not.toHaveBeenCalled();
+
+                mockEvent.which = 13;
+                controller.updateName(mockEvent);
+                mockMutationCapability.mutate.mostRecentCall.args[0](mockDomainObject.model);
```

Comment:
Move this below `expect(mockMutationCapability.mutate).toHaveBeenCalled();`. If the expectation were to fail, we'd get a clear error message `"Error: Expected spy mutation.mutate to have been called."` instead of a `"TypeError: Cannot read property '0' of undefined"`.